tests/installation/first_boot.pm

```diff
@@ -79,11 +79,17 @@ sub post_fail_hook {
     # save a screenshot before trying further measures which might fail
     save_screenshot;
     # if we found a shell, we do not need the memory dump
-    if (!(match_has_tag('emergency-shell') or match_has_tag('emergency-mode'))) {
-        die "save_memory_dump is temporarily unavailable, see https://progress.opensuse.org/issues/19390";
-        die "save_memory_dump not implemented, no way to save memory_dump" unless check_var('BACKEND', 'qemu');
-        diag 'Save memory dump to debug bootup problems, e.g. for bsc#1005313';
-        save_memory_dump;
+  MEMORY_DUMP: {
+        if (!(match_has_tag('emergency-shell') or match_has_tag('emergency-mode'))) {
+            diag 'save_memory_dump is temporarily unavailable, see https://progress.opensuse.org/issues/19390';
+            last MEMORY_DUMP;
```

Comment:
It look like this like will break this code section unconditionally and that's not what we want here. As an alternative to labels we could use function and return statements.

Reply:
But that's the point. In before it was just "die". I would call that "unconditional breaking" as well ;)

Reply:
I do not get it, you've refactored code which was not executed and won't be executed and instead of fancy goto style labels you could simply return. As a side note, maybe we should record soft-failure here and not diag. WDYT?

Reply:
Why not just replace "die" with "diag" and comment out non-reachable code with poo# reference?
Labels here are making code harder to read (at least for me).

Reply:
I'm with @kravciak here. Except commented code as I would remove it, we have VCS, so can find old code. And we have progress ticket which will remind us to re-enable previous behavior.